regular_expressions_boundaries_anchors/regular_expressions_boundaries_anchors.md

```diff
@@ -0,0 +1,419 @@
+
+</div>
+
+### Except: Another Use of the Caret 
+
+On the previous page, we warned you about a potential pitfall:
+
+> To correctly use `^` as the start-of-string token, you must not enclose it in square brackets.  If `^` appears inside square brackets, it will be intepreted as doing something quite different, as you are about to discover.
+
+Let's look at three ways we can use the caret (`^`).
+
+Start-of-string Caret
+-----
+
+If we want to indicate the start of a string, we add the caret at the beginning of our regular expression, being sure to not enclose it in square brackets.  We see an example of this in the regular expression `^(?:WT|PBMC)\-\d{5,6}`, which includes a start of string character followed by a non-capturing group with two alternatives, a hyphen, and then five to six digits.
+
+Except-for Caret
+------
+
+We can also use the caret symbol in an "except for" clause in our regular expression.  For example, what if at a given position in a pattern, any character is possible except for a hash mark (#)?  We can say "except for" by using `^` as the first character within square brackets: `[^#]`.  
+
+We can add multiple characters that should not appear within the square brackets.  If any character except for  #, 5, ., and % are permitted at a given location in a string, we could express that with the regular expression list `[^#5\.%]`. For example, `[ABXYZ]` means "any of these: A, B, X, Y, or Z", while `[^ABXYZ]` means "anything that **isn't** one of these: A, B, X, Y, and Z."  
+
+<div class = "important">
+<b style="color: rgb(var(--color-highlight));">Important note</b><br>
+
+To use `^` as "except for", it has to immediately follow an opening square bracket, and be followed by the characters it is indicating aren't acceptable.  You can't, for example, say "all the digits except for 7" by typing `\d^7` or `[\d^7]`.
+
+</div>
+
+Literal Caret
+---
+
+If you're looking for the actual symbol `^` in your text, you'll have to **escape** it by adding a backslash right before the caret: `\^`.  As a reminder, that backslash is useful for all **metacharacters** in regular expressions.  You'd do the same thing for the end-of-string character: `\$` is how you look for a literal dollar sign / money symbol.
+
+### End-of-string Anchoring 
+
+As a reminder, the symbol for end-of-string in a regular expression is `$`.  We can use this symbol to look for specific patterns at the end of strings.  For example, we could look for specific typos that tend to appear at the end of strings: periods that follow single or double quotation marks instead of being enclosed within the quotation.  Here are some examples of acceptable grammar and typos:
+
+**Acceptable**
+
+* stated, "I want to withdraw from the study."
+* not clear on pre-registration next steps.'
```

Comment:
It isn't quite right to end a sentence in `.'` since single quotes only ever appear inside double quotes (see rule 7 and 8b here: https://www.grammarbook.com/punctuation/quotes.asp)

I kind of feel bad for even knowing that, let alone looking it up and being right and sharing the link....

Actually, I think that as wonderful as this is and as honored as I am that I might have drilled this punctuation rule into your head, it is actually not a great example because it is particularly un-translatable... Even other languages using the same alphabet have different punctuation rules. I'll think about another typo I can try to suggest.

Reply:
You did make me proud!!

Reply:
@drelliche Can we just pretend this is a reasonable thing to search for?  I could also just search for the double quote followed by a period, but I like to complicate things by giving an example where there are two typos we're looking for.